pkg/security/probe/kfilters/kfilters_bpf.go

```diff
@@ -69,6 +70,10 @@ func (e *arrayEntry) Key() interface{} {
 	}
 }
 
+func (e *arrayEntry) GetTableName() string {
```

Comment:
I would add a `Type` field instead of using the table name

Reply:
since arrayEntry is so generic though, would a type field make sense for other uses of that type? I use the concept of "type" later in the call path when we're solidly in the realm of "approvers" and not "generic kernel array entry"
```
func getApproverType(approverTableName string) string {
	approverType := "flag"

	if approverTableName == kfilters.BasenameApproverKernelMapName {
		approverType = "basename"
	}

	return approverType
}
```